src/main/java/seedu/parser/MainParser.java

```diff
@@ -69,6 +72,10 @@ private Command parseAddContact(String userInput) {
         contactParser = addContactParser;
         try {
             String[] details = contactParser.parseContactDetails(userInput);
+            //check if name is specified in input
+            if (details[NAME_INDEX] == null) {
+                throw new MissingDetailException();
+            }
```

Comment:
Hi @ng-andre , I understand what you are trying to do here. However, the `MissingDetailException` here appears to be for flags with nothing specified, ie `add -n -g github`. As such, the given FailedCommand would result in a not-so-suitable message being displayed. To improve this, I would suggest making a new exception, followed by a suitable error message in TextUi.

Reply:
Will update this in my next commit, thanks!